test/src/core/encoding.spec.js

```diff
@@ -0,0 +1,496 @@
+/*****************************************************************************
+ * Copyright (c) 2021 Bosch.IO GmbH
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ ****************************************************************************/
+
+const jsonata = require('jsonata');
+ 
+const {
+    ENCODING_TOPIC,
+    ROUTING_TOPIC,
+    ENCODING_ENCODER_CATEGORY,
+    ENCODING_ENCODER_MINMAX,
+    ENCODING_ENCODER_DELTA,
+    ENCODING_ENCODER_THROUGH,
+    ENCODING_TYPE_NUMBER,
+    ENCODING_TYPE_STRING,
+    ENCODING_TYPE_OBJECT,
+    ENCODING_TYPE_BOOLEAN,
+    DEFAULT_FEATURE_TTL_MS
+} = require('../../../src/core/constants');
+
+const Logger = require('../../../src/core/util/logger');
+const InstanceManager = require('../../../src/core/instanceManager');
+const equals = require('../../../src/core/util/equals');
+const jsonQuery = require('../../../src/core/util/jsonQuery');
+const clone = require('../../../src/core/util/clone');
+const ProtocolGateway = require('../../../src/core/protocolGateway');
+const Encoding = require('../../../src/core/encoding');
+const NullLogger = require('../../helpers/null.logger');
+const Instance = require('../../../src/core/instance');
+
+const { MqttProtocolAdapter } = require('../../../src/core/util/mqttClient');
+
+describe('core.encoding', () => {
+
+    function createInstance(type, instanceId) {
+        return new Instance(type, instanceId, new NullLogger());
+    }
+
+    function createEncoding() {
+        return new Encoding(ProtocolGateway.createDefaultConfiguration([ MqttProtocolAdapter.createDefaultConfiguration(true, '')]));
+    }
+
+    const nowMs = Date(); //  1618394584450;
+    const i = createInstance('test', '4711');
+    i.updateFeatureAt(1, null, 'test', nowMs, 0, DEFAULT_FEATURE_TTL_MS, nowMs);
+    const enc = createEncoding();
+
+    describe('Start Encoder ', () => {
+        it('should not be null ', () => {
+            expect(enc.start()).not.toBeNull();;
+        });
+    });
+    
+
+    describe('__encodeValue with encoder: null should return null ', () => {
+        it('encoder == null ', () => {
+
+            const value = 'test';
+            const metaFeature = {
+                idx: 1,
+                description: '1',
+                encoding: {
+                    encoder: null,
+                    type: 'number'
+                } 
+            }     
+            expect(enc.__encodeValue(value, metaFeature, i.getFeatureAt(1))).toBe(null);
+        });
+
+        it('encoder == ENCODING_ENCODER_THROUGH happy path', () => {
```

Comment:
(This comment is valid for all tests in this file)
Test names / descriptions are not aligned with the other nodejs tests.
This could be something like 'Should return the correct encoded values using ENCODING_ENCODER_THROUGH'